Update to buildroot 2018.11

ski7777 opened the pull request:
I hope this will work. The linaro toolchain was updated so the really old kernel 4.1 won´t build anymore, so I had to downgrade the toolchain again...: https://github.com/ftCommunity/buildroot/commit/2a5515c496f0366b49dbc1e330c22cf025903001

All in all it´s a bit dirty but I hope somebody can help me updating the kernel.

Raphael

P.S: I will upload binaries later

rkunze commented:
> I hope this will work. 

Well, try it out, then you *know* if it works ;-)

>The linaro toolchain was updated so the really old kernel 4.1 won´t build anymore, so I had to downgrade the toolchain again

We should upgrade the kernel instead. My choice would be 4.14 (that's the current longterm kernel, and it is supported by TI)

> I hope somebody can help me updating the kernel.

First steps for updating:
* Change BR2_LINUX_KERNEL_CUSTOM_REPO_VERSION to "ti-linux-4.14.y"
* Try to build, check for failures
* Fix the errors  (I'd expect some in the DTS)

I'm a bit short on time right now, but I'll try to help - just drop me a note that tells me where you got stuck (or better yet: Push the breaking change to this PR, and explain in the commit message what exactly went wrong).

ski7777 commented:
Sorry for the force push. Currently I´m building. DTS build worked flawlessly. I think we will encounter boot problems. I will check this in the next few hours

ski7777 commented:
```
Starting kernel ...

[    0.000989] timer_probe: no matching timers found
[    1.386146] Unable to handle kernel NULL pointer dereference at virtual address 00000000
[    1.395605] pgd = c0004000
[    1.398463] [00000000] *pgd=00000000
[    1.402619] Internal error: Oops: 5 [#1] PREEMPT ARM
[    1.407833] Modules linked in:
[    1.411057] CPU: 0 PID: 1 Comm: swapper Not tainted 4.14.79 #2
[    1.417168] Hardware name: Generic AM33XX (Flattened Device Tree)
[    1.423554] task: cd83ad00 task.stack: cd83c000
[    1.428319] PC is at kim_probe+0x78/0x214
[    1.432539] LR is at __lockdep_init_map+0x38/0x214
[    1.437560] pc : [<c0592494>]    lr : [<c016ff64>]    psr: 60000113
[    1.444126] sp : cd83de60  ip : 00000000  fp : c0c64b98
[    1.449599] r10: c0c004c8  r9 : 00000000  r8 : 00000000
[    1.455075] r7 : c1620344  r6 : c0ea7c5c  r5 : cd98bc00  r4 : cdd16400
[    1.461914] r3 : cdf75800  r2 : c0591d60  r1 : c0ac83d8  r0 : 00000000
[    1.468756] Flags: nZCv  IRQs on  FIQs on  Mode SVC_32  ISA ARM  Segment none
[    1.476232] Control: 10c5387d  Table: 80004019  DAC: 00000051
[    1.482253] Process swapper (pid: 1, stack limit = 0xcd83c210)
[    1.488364] Stack: (0xcd83de60 to 0xcd83e000)
[    1.492942] de60: c161ff54 c0ea7c5c cd98bc10 ffffffed c0ea7c5c fffffdfb c0ea7c5c c056a590
[    1.501517] de80: cd98bc10 c161ff54 00000000 00000000 c0ea7c5c c056872c cd98bc10 c0ea7c5c
[    1.510093] dea0: cd98bc44 00000000 000000fd c0c52850 c0c004c8 c05688b4 00000000 c0ea7c5c
[    1.518669] dec0: c05687fc c0566cd8 cd886ea0 cd9847d0 c0ea7c5c cdf71880 c0ea5fd8 c0567cd0
[    1.527244] dee0: c0ac83d8 00000000 c0c38fbc c0ea7c5c 00000000 c0c38fbc c0ee9000 c0569740
[    1.535819] df00: ffffe000 00000000 c0c38fbc c0101c90 00000000 cfdffca7 00000000 cfdffc9e
[    1.544394] df20: c0bb5210 000000fd cfdffcac c0153688 c0b40dd8 00000006 00000006 00000000
[    1.552969] df40: cfdffca7 00000000 c0c004c8 00000007 c0c5286c c0ee9000 00000007 c0c52870
[    1.561543] df60: c0ee9000 c0ee9000 000000fd c0c00d6c 00000006 00000006 00000000 c0c004c8
[    1.570118] df80: 00000000 00000000 c08e39bc 00000000 00000000 00000000 00000000 00000000
[    1.578691] dfa0: 00000000 c08e39c4 00000000 c0107e94 00000000 00000000 00000000 00000000
[    1.587266] dfc0: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
[    1.595839] dfe0: 00000000 00000000 00000000 00000000 00000013 00000000 00000000 00000000
[    1.604439] [<c0592494>] (kim_probe) from [<c056a590>] (platform_drv_probe+0x50/0xac)
[    1.612656] [<c056a590>] (platform_drv_probe) from [<c056872c>] (driver_probe_device+0x208/0x2d8)
[    1.621959] [<c056872c>] (driver_probe_device) from [<c05688b4>] (__driver_attach+0xb8/0xbc)
[    1.630808] [<c05688b4>] (__driver_attach) from [<c0566cd8>] (bus_for_each_dev+0x78/0x9c)
[    1.639383] [<c0566cd8>] (bus_for_each_dev) from [<c0567cd0>] (bus_add_driver+0x198/0x1fc)
[    1.648048] [<c0567cd0>] (bus_add_driver) from [<c0569740>] (driver_register+0x78/0xf4)
[    1.656444] [<c0569740>] (driver_register) from [<c0101c90>] (do_one_initcall+0x3c/0x17c)
[    1.665031] [<c0101c90>] (do_one_initcall) from [<c0c00d6c>] (kernel_init_freeable+0x100/0x1b4)
[    1.674161] [<c0c00d6c>] (kernel_init_freeable) from [<c08e39c4>] (kernel_init+0x8/0x110)
[    1.682744] [<c08e39c4>] (kernel_init) from [<c0107e94>] (ret_from_fork+0x14/0x20)
[    1.690684] Code: 1a000053 e5943094 e59f1160 e58340ec (e5980000) 
[    1.697282] ---[ end trace b463238909e54117 ]---
[    1.702260] Kernel panic - not syncing: Attempted to kill init! exitcode=0x0000000b
[    1.702260] 
[    1.711852] ---[ end Kernel panic - not syncing: Attempted to kill init! exitcode=0x0000000b
[    1.711852] 

```
Here are the binaries: https://github.com/ski7777/ftcommunity-TXT/releases/tag/v0.9.5-dev.snapshot-20181206

Maybe you know what to do now and can push me in the right direction.

Raphael

ski7777 commented:
```
Starting kernel ...

[    0.000988] timer_probe: no matching timers found
[    1.398543] mtdoops: mtd device (mtddev=name/number) must be supplied
[    1.420300] spidev spi1.0: buggy DT: spidev listed directly in DT
[    1.735103] omap_voltage_late_init: Voltage driver support not added
/bin/busybox: error while loading shared libraries: libresolv.so.2: cannot open shared object file: No such file or directory
[    2.260513] Kernel panic - not syncing: Attempted to kill init! exitcode=0x00007f00
[    2.260513] 
[    2.270301] ---[ end Kernel panic - not syncing: Attempted to kill init! exitcode=0x00007f00
[    2.270301] 
[    2.281178]  mmcblk1: p1 p2

```

Here are the binaries: https://github.com/ski7777/ftcommunity-TXT/releases/tag/0.9.5-rc%2Bv0.9.5-dev.snapshot-20181207

I will check what is going on with the missing lib. I´m quite sure I need t change something in the initramfs config.

Raphael

ski7777 commented:
Done.

Next problem:
```
Starting kernel ...

[    0.000985] timer_probe: no matching timers found
[    1.404497] mtdoops: mtd device (mtddev=name/number) must be supplied
[    1.425877] spidev spi1.0: buggy DT: spidev listed directly in DT
[    1.740600] omap_voltage_late_init: Voltage driver support not added
mount: mounting /dev/mmcblk0p1 on /media/sdcard/boot failed: No such file or directory
mount: mounting /media/sdcard/boot/rootfs.img on /newroot failed: No such file or directory
cp: can't stat '/newroot/usr/lib/libfuse.so.2': No such file or directory
cp: can't stat '/newroot/lib/libudev.so.1': No such file or directory
cp: can't stat '/newroot/lib/librt.so.1': No such file or directory
cp: can't stat '/newroot/lib/libpthread.so.0': No such file or directory
cp: can't stat '/newroot/usr/lib/libext2fs.so.2': No such file or directory
cp: can't stat '/newroot/usr/lib/libcom_err.so.2': No such file or directory
cp: can't stat '/newroot/lib/libblkid.so.1': No such file or directory
cp: can't stat '/newroot/lib/libuuid.so.1': No such file or directory
cp: can't stat '/newroot/usr/lib/libe2p.so.2': No such file or directory
cp: can't stat '/newroot/usr/lib/libpng16.so.16': No such file or directory
cp: can't stat '/newroot/usr/lib/libjpeg.so.8': No such file or directory
cp: can't stat '/newroot/usr/lib/libgif.so.7': No such file or directory
cp: can't stat '/newroot/usr/lib/libz.so.1': No such file or directory
cp: can't stat '/newroot/lib/libm.so.6': No such file or directory
cp: can't stat '/newroot/sbin/mkfs.fat': No such file or directory
cp: can't stat '/newroot/sbin/mke2fs': No such file or directory
cp: can't stat '/newroot/usr/bin/psplash': No such file or directory
cp: can't stat '/newroot/usr/bin/psplash-write': No such file or directory
cp: can't stat '/newroot/usr/bin/fbv': No such file or directory
umount: can't unmount /newroot: Invalid argument
/repart_Hh�'��X��.]X��'/media/sdcard/boot/uImage': No such file or directory
/repartition-sd-card: line 34: psplash-write: not found
/repartition-sd-card: line 36: psplash-write: not found
/repartition-sd-card: line 37: psplash-write: not found
failed to repartition the SD card. Dropping to a shell
/repartition-sd-card: line 35: psplash: not found
/bin/sh: can't access tty; job control turned off

```

So I need to check where to find the sd card. Maybe some things in the dts are wrong.

binaries: https://github.com/ski7777/ftcommunity-TXT/releases/tag/v0.9.5-dev.snapshot-20181206 (upload still pending. May take a while)

Raphael

ski7777 commented:
Now I executed `ls -la /dev` in the initramfs:
```
total 0
drwxr-xr-x    2 0        0                0 Dec  7  2018 .
drwxr-xr-x   10 0        0                0 Jan  1 00:00 ..
crw-------    1 0        0           5,   1 Dec  7  2018 console
crw-rw-rw-    1 0        0           1,   3 Dec  7  2018 null
crw-------    1 0        0         252,   0 Dec  7  2018 ttyO0
```

So the sdcard is not detected. Do you know how to fix this?

Raphael

rkunze commented:
> mtd is the nand flash.
Yes, you're right - got it confused with mmc.
But I'd nonetheless try and check the boot log for mmc related messages. There'll probably be some somewhere.

ski7777 commented:
No. That´s all I have:
```
[    0.000985] timer_probe: no matching timers found
[    1.401762] mtdoops: mtd device (mtddev=name/number) must be supplied
[    1.423543] spidev spi1.0: buggy DT: spidev listed directly in DT
[    1.740210] omap_voltage_late_init: Voltage driver support not added
```

rkunze commented:
Intermediate result: I have a build with a slightly updated DTS (basically reordered the DTS a bit for esaier comparison with similar boards from the mainline kernel). This build has the same symptoms as yours (MMC card is not found). Reason is that the MMC card ist detected as mmcblk1 instead of mmcblk0. I'm currently investigating why.

If you want to integrate my changes, pull from https://github.com/rkunze/ftcommunity-TXT/tree/update-buildroot-2018.11

ski7777 commented:
I´m currently thinking about integrating this:
```
diff --git a/configs/fischertechnik_TXT_defconfig b/configs/fischertechnik_TXT_defconfig
index 3d1c3001fc..0b4f2ca4a6 100644
--- a/configs/fischertechnik_TXT_defconfig
+++ b/configs/fischertechnik_TXT_defconfig
@@ -59,8 +59,8 @@ BR2_PACKAGE_HOST_LINUX_HEADERS_CUSTOM_4_14=y
 # kernel
 BR2_LINUX_KERNEL=y
 BR2_LINUX_KERNEL_CUSTOM_GIT=y
-BR2_LINUX_KERNEL_CUSTOM_REPO_URL="git://git.ti.com/ti-linux-kernel/ti-linux-kernel.git"
-BR2_LINUX_KERNEL_CUSTOM_REPO_VERSION="ti-linux-4.14.y"
+BR2_LINUX_KERNEL_CUSTOM_REPO_URL="git://git.ti.com/processor-sdk/processor-sdk-linux.git"
+BR2_LINUX_KERNEL_CUSTOM_REPO_VERSION="processor-sdk-linux-4.14.y"
 BR2_LINUX_KERNEL_USE_CUSTOM_CONFIG=y
 BR2_LINUX_KERNEL_CUSTOM_CONFIG_FILE="$(BR2_EXTERNAL_FTCOMMUNITY_TXT_PATH)/board/fischertechnik/TXT/tisdk_am335x-fischertechnik_txt_defconfig"
 BR2_LINUX_KERNEL_UIMAGE=y
```

I read somewhere that this repository should be used but I don´t really know.

ski7777 commented:
```
[    0.000991] timer_probe: no matching timers found
[    1.408005] mtdoops: mtd device (mtddev=name/number) must be supplied
[    1.430485] spidev spi1.0: buggy DT: spidev listed directly in DT
[    1.745793] omap_voltage_late_init: Voltage driver support not added
[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 4.14.79 (raphael@raphael-laptop) (gcc version 8.2.1 20180802 (GNU Toolchain for the A-profile Architecture 8.2-2018-08 (arm-rel-8.23))) #8 PREEMPT Sat Dec 8 14:02:03 CET 2018
[    0.000000] CPU: ARMv7 Processor [413fc082] revision 2 (ARMv7), cr=10c5387d
[    0.000000] CPU: PIPT / VIPT nonaliasing data cache, VIPT aliasing instruction cache
[    0.000000] OF: fdt: Machine model: Knobloch TXT AM3352
[    0.000000] Memory policy: Data cache writeback
[    0.000000] cma: Reserved 16 MiB at 0x8e000000
[    0.000000] On node 0 totalpages: 65024
[    0.000000] free_area_init_node: node 0, pgdat c0edf864, node_mem_map cfbf1000
[    0.000000]   Normal zone: 508 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 65024 pages, LIFO batch:15
[    0.000000] CPU: All CPU(s) started in SVC mode.
[    0.000000] AM335X ES2.1 (neon)
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 64516
[    0.000000] Kernel command line: fbtft_device.name=txt_ili9341 fbtft_device.fps=10 console=ttyO0,115200 root=/dev/mmcblk0p2 rw rootwait quiet
[    0.000000] PID hash table entries: 1024 (order: 0, 4096 bytes)
[    0.000000] Dentry cache hash table entries: 32768 (order: 5, 131072 bytes)
[    0.000000] Inode-cache hash table entries: 16384 (order: 4, 65536 bytes)
[    0.000000] Memory: 219460K/260096K available (8192K kernel code, 924K rwdata, 2776K rodata, 2048K init, 7469K bss, 24252K reserved, 16384K cma-reserved, 0K highmem)
[    0.000000] Virtual kernel memory layout:
[    0.000000]     vector  : 0xffff0000 - 0xffff1000   (   4 kB)
[    0.000000]     fixmap  : 0xffc00000 - 0xfff00000   (3072 kB)
[    0.000000]     vmalloc : 0xd0000000 - 0xff800000   ( 760 MB)
[    0.000000]     lowmem  : 0xc0000000 - 0xcfe00000   ( 254 MB)
[    0.000000]     pkmap   : 0xbfe00000 - 0xc0000000   (   2 MB)
[    0.000000]     modules : 0xbf000000 - 0xbfe00000   (  14 MB)
[    0.000000]       .text : 0xc0008000 - 0xc0900000   (9184 kB)
[    0.000000]       .init : 0xc0c00000 - 0xc0e00000   (2048 kB)
[    0.000000]       .data : 0xc0e00000 - 0xc0ee711c   ( 925 kB)
[    0.000000]        .bss : 0xc0ee9000 - 0xc16347d4   (7470 kB)
[    0.000000] Running RCU self tests
[    0.000000] Preemptible hierarchical RCU implementation.
[    0.000000]  RCU lockdep checking is enabled.
[    0.000000]  Tasks RCU enabled.
[    0.000000] NR_IRQS: 16, nr_irqs: 16, preallocated irqs: 16
[    0.000000] IRQ: Found an INTC at 0xfa200000 (revision 5.0) with 128 interrupts
[    0.000000] OMAP clockevent source: timer2 at 24000000 Hz
[    0.000024] sched_clock: 32 bits at 24MHz, resolution 41ns, wraps every 89478484971ns
[    0.000058] clocksource: timer1: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 79635851949 ns
[    0.000125] OMAP clocksource: timer1 at 24000000 Hz
[    0.000991] timer_probe: no matching timers found
[    0.002205] Console: colour dummy device 80x30
[    0.002263] Lock dependency validator: Copyright (c) 2006 Red Hat, Inc., Ingo Molnar
[    0.002278] ... MAX_LOCKDEP_SUBCLASSES:  8
[    0.002292] ... MAX_LOCK_DEPTH:          48
[    0.002305] ... MAX_LOCKDEP_KEYS:        8191
[    0.002319] ... CLASSHASH_SIZE:          4096
[    0.002332] ... MAX_LOCKDEP_ENTRIES:     32768
[    0.002344] ... MAX_LOCKDEP_CHAINS:      65536
[    0.002357] ... CHAINHASH_SIZE:          32768
[    0.002370]  memory used by lock dependency info: 4655 kB
[    0.002383]  per task-struct memory footprint: 1536 bytes
[    0.002432] Calibrating delay loop... 594.73 BogoMIPS (lpj=1486848)
[    0.057951] pid_max: default: 32768 minimum: 301
[    0.058240] Security Framework initialized
[    0.058388] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.058411] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.061488] CPU: Testing write buffer coherency: ok
[    0.061718] CPU0: Spectre v2: using BPIALL workaround
[    0.065450] Setting up static identity map for 0x80100000 - 0x80100078
[    0.066153] Hierarchical SRCU implementation.
[    0.072179] devtmpfs: initialized
[    0.114062] random: get_random_u32 called from bucket_table_alloc+0xe4/0x240 with crng_init=0
[    0.114588] VFP support v0.3: implementor 41 architecture 3 part 30 variant c rev 3
[    0.117883] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 9556302231375000 ns
[    0.118115] futex hash table entries: 256 (order: 1, 11264 bytes)
[    0.120022] pinctrl core: initialized pinctrl subsystem
[    0.131864] NET: Registered protocol family 16
[    0.142327] DMA: preallocated 256 KiB pool for atomic coherent allocations
[    0.214950] omap_hwmod: debugss: _wait_target_disable failed
[    0.279570] cpuidle: using governor menu
[    0.317123] gpio gpiochip0: (gpio): added GPIO chardev (254:0)
[    0.318452] gpiochip_setup_dev: registered GPIOs 0 to 31 on device: gpiochip0 (gpio)
[    0.318738] OMAP GPIO hardware version 0.1
[    0.324465] gpio gpiochip1: (gpio): added GPIO chardev (254:1)
[    0.325118] gpiochip_setup_dev: registered GPIOs 32 to 63 on device: gpiochip1 (gpio)
[    0.330704] gpio gpiochip2: (gpio): added GPIO chardev (254:2)
[    0.331313] gpiochip_setup_dev: registered GPIOs 64 to 95 on device: gpiochip2 (gpio)
[    0.336855] gpio gpiochip3: (gpio): added GPIO chardev (254:3)
[    0.337495] gpiochip_setup_dev: registered GPIOs 96 to 127 on device: gpiochip3 (gpio)
[    0.378791] omap-gpmc 50000000.gpmc: could not find pctldev for node /ocp/l4_wkup@44c00000/scm@210000/pinmux@800/nandflash_pins_s0, deferring probe
[    0.392019] No ATAGs?
[    0.392055] hw-breakpoint: debug architecture 0x4 unsupported.
[    0.392565] omap4_sram_init:Unable to allocate sram needed to handle errata I688
[    0.392592] omap4_sram_init:Unable to get sram pool needed to handle errata I688
[    0.398447] fbtft_device: GPIOS used by 'txt_ili9341':
[    0.398478] fbtft_device: 'led' = GPIO7
[    0.398642] platform fb_ili9341_txt.0: fb_ili9341_txt id=0 pdata? yes
[    0.612435] edma 49000000.edma: TI EDMA DMA engine driver
[    0.616486] of_get_named_gpiod_flags: can't parse 'gpio' property of node '/fixedregulator@0[0]'
[    0.619647] of_get_named_gpiod_flags: parsed 'gpio' property of node '/fixedregulator@1[0]' - status (0)
[    0.713176] SCSI subsystem initialized
[    0.716183] usbcore: registered new interface driver usbfs
[    0.717094] usbcore: registered new interface driver hub
[    0.720648] usbcore: registered new device driver usb
[    0.722416] omap_i2c 44e0b000.i2c: could not find pctldev for node /ocp/l4_wkup@44c00000/scm@210000/pinmux@800/pinmux_i2c0_pins, deferring probe
[    0.722581] omap_i2c 4802a000.i2c: could not find pctldev for node /ocp/l4_wkup@44c00000/scm@210000/pinmux@800/pinmux_i2c1_pins, deferring probe
[    0.724843] Linux video capture interface: v2.00
[    0.737627] Bluetooth: Core ver 2.22
[    0.738729] NET: Registered protocol family 31
[    0.738752] Bluetooth: HCI device and connection manager initialized
[    0.738949] Bluetooth: HCI socket layer initialized
[    0.738985] Bluetooth: L2CAP socket layer initialized
[    0.739226] Bluetooth: SCO socket layer initialized
[    0.744473] clocksource: Switched to clocksource timer1
[    0.960444] VFS: Disk quotas dquot_6.6.0
[    0.960626] VFS: Dquot-cache hash table entries: 1024 (order 0, 4096 bytes)
[    1.049049] NET: Registered protocol family 2
[    1.053419] TCP established hash table entries: 2048 (order: 1, 8192 bytes)
[    1.053514] TCP bind hash table entries: 2048 (order: 4, 73728 bytes)
[    1.053987] TCP: Hash tables configured (established 2048 bind 2048)
[    1.055081] UDP hash table entries: 256 (order: 2, 20480 bytes)
[    1.055242] UDP-Lite hash table entries: 256 (order: 2, 20480 bytes)
[    1.055811] NET: Registered protocol family 1
[    1.057777] RPC: Registered named UNIX socket transport module.
[    1.057827] RPC: Registered udp transport module.
[    1.057845] RPC: Registered tcp transport module.
[    1.057862] RPC: Registered tcp NFSv4.1 backchannel transport module.
[    1.173092] hw perfevents: no interrupt-affinity property for /pmu, guessing.
[    1.173496] hw perfevents: enabled with armv7_cortex_a8 PMU driver, 5 counters available
[    1.190322] workingset: timestamp_bits=30 max_order=16 bucket_order=0
[    1.192099] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    1.195704] NFS: Registering the id_resolver key type
[    1.196213] Key type id_resolver registered
[    1.196261] Key type id_legacy registered
[    1.197810] jffs2: version 2.2. (NAND) (SUMMARY)  © 2001-2006 Red Hat, Inc.
[    1.200658] fuse init (API version 7.26)
[    1.224246] io scheduler noop registered
[    1.224278] io scheduler deadline registered
[    1.224427] io scheduler cfq registered (default)
[    1.225029] io scheduler mq-deadline registered
[    1.225048] io scheduler kyber registered
[    1.231206] pinctrl-single 44e10800.pinmux: 142 pins at pa f9e10800 size 568
[    1.243649] Serial: 8250/16550 driver, 4 ports, IRQ sharing enabled
[    1.257265] omap_uart 44e09000.serial: no wakeirq for uart0
[    1.257318] of_get_named_gpiod_flags: can't parse 'rts-gpio' property of node '/ocp/serial@44e09000[0]'
[    1.259126] 44e09000.serial: ttyO0 at MMIO 0x44e09000 (irq = 30, base_baud = 3000000) is a OMAP UART0
[    1.266500] console [ttyO0] enabled
[    1.269103] omap_uart 48024000.serial: no wakeirq for uart2
[    1.269148] of_get_named_gpiod_flags: can't parse 'rts-gpio' property of node '/ocp/serial@48024000[0]'
[    1.269902] 48024000.serial: ttyO2 at MMIO 0x48024000 (irq = 31, base_baud = 3000000) is a OMAP UART2
[    1.272966] omap_uart 481a6000.serial: no wakeirq for uart3
[    1.273008] of_get_named_gpiod_flags: can't parse 'rts-gpio' property of node '/ocp/serial@481a6000[0]'
[    1.273492] 481a6000.serial: ttyO3 at MMIO 0x481a6000 (irq = 32, base_baud = 3000000) is a OMAP UART3
[    1.274860] random: fast init done
[    1.281592] omap_rng 48310000.rng: Random Number Generator ver. 20
[    1.346896] brd: module loaded
[    1.393505] loop: module loaded
[    1.408005] mtdoops: mtd device (mtddev=name/number) must be supplied
[    1.430485] spidev spi1.0: buggy DT: spidev listed directly in DT
[    1.437587] ------------[ cut here ]------------
[    1.437632] WARNING: CPU: 0 PID: 1 at drivers/spi/spidev.c:730 spidev_probe+0x1ac/0x1ec
[    1.437650] Modules linked in:
[    1.437691] CPU: 0 PID: 1 Comm: swapper Not tainted 4.14.79 #8
[    1.437708] Hardware name: Generic AM33XX (Flattened Device Tree)
[    1.437751] [<c010e7b0>] (unwind_backtrace) from [<c010c14c>] (show_stack+0x10/0x14)
[    1.437794] [<c010c14c>] (show_stack) from [<c0131f78>] (__warn+0xc8/0x100)
[    1.437825] [<c0131f78>] (__warn) from [<c013205c>] (warn_slowpath_null+0x20/0x28)
[    1.437853] [<c013205c>] (warn_slowpath_null) from [<c05f2448>] (spidev_probe+0x1ac/0x1ec)
[    1.437895] [<c05f2448>] (spidev_probe) from [<c05ee5c8>] (spi_drv_probe+0x7c/0xac)
[    1.437931] [<c05ee5c8>] (spi_drv_probe) from [<c056872c>] (driver_probe_device+0x208/0x2d8)
[    1.437957] [<c056872c>] (driver_probe_device) from [<c0566d80>] (bus_for_each_drv+0x70/0x94)
[    1.437983] [<c0566d80>] (bus_for_each_drv) from [<c0568440>] (__device_attach+0xb0/0x118)
[    1.438007] [<c0568440>] (__device_attach) from [<c0567a34>] (bus_probe_device+0x88/0x90)
[    1.438032] [<c0567a34>] (bus_probe_device) from [<c0565e10>] (device_add+0x338/0x5a0)
[    1.438055] [<c0565e10>] (device_add) from [<c05ef5c4>] (spi_add_device+0x90/0x130)
[    1.438080] [<c05ef5c4>] (spi_add_device) from [<c05efe60>] (spi_register_controller+0x304/0xa10)
[    1.438105] [<c05efe60>] (spi_register_controller) from [<c05f059c>] (devm_spi_register_controller+0x30/0x70)
[    1.438131] [<c05f059c>] (devm_spi_register_controller) from [<c05f50a4>] (omap2_mcspi_probe+0x27c/0x324)
[    1.438161] [<c05f50a4>] (omap2_mcspi_probe) from [<c056a590>] (platform_drv_probe+0x50/0xac)
[    1.438188] [<c056a590>] (platform_drv_probe) from [<c056872c>] (driver_probe_device+0x208/0x2d8)
[    1.438214] [<c056872c>] (driver_probe_device) from [<c05688b4>] (__driver_attach+0xb8/0xbc)
[    1.438239] [<c05688b4>] (__driver_attach) from [<c0566cd8>] (bus_for_each_dev+0x78/0x9c)
[    1.438263] [<c0566cd8>] (bus_for_each_dev) from [<c0567cd0>] (bus_add_driver+0x198/0x1fc)
[    1.438289] [<c0567cd0>] (bus_add_driver) from [<c0569740>] (driver_register+0x78/0xf4)
[    1.438316] [<c0569740>] (driver_register) from [<c0101c90>] (do_one_initcall+0x3c/0x17c)
[    1.438352] [<c0101c90>] (do_one_initcall) from [<c0c00d6c>] (kernel_init_freeable+0x100/0x1b4)
[    1.438386] [<c0c00d6c>] (kernel_init_freeable) from [<c08e3a64>] (kernel_init+0x8/0x110)
[    1.438420] [<c08e3a64>] (kernel_init) from [<c0107e94>] (ret_from_fork+0x14/0x20)
[    1.438520] ---[ end trace cd521035b6e90be9 ]---
[    1.447659] mdio_bus fixed-0: GPIO lookup for consumer reset
[    1.447694] mdio_bus fixed-0: using lookup tables for GPIO lookup
[    1.447839] mdio_bus fixed-0: lookup for GPIO reset failed
[    1.448168] libphy: Fixed MDIO Bus: probed
[    1.451763] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    1.452277] usbcore: registered new interface driver cdc_acm
[    1.452295] cdc_acm: USB Abstract Control Model driver for USB modems and ISDN adapters
[    1.452656] usbcore: registered new interface driver cdc_wdm
[    1.453690] usbcore: registered new interface driver usb-storage
[    1.454147] usbcore: registered new interface driver usbtest
[    1.462749] input: ti-tsc as /devices/platform/ocp/44e0d000.tscadc/44e0d000.tscadc:tsc/input/input0
[    1.475844] omap_rtc 44e3e000.rtc: registered as rtc0
[    1.477205] i2c /dev entries driver
[    1.478853] IR NEC protocol handler initialized
[    1.478881] IR RC5(x/sz) protocol handler initialized
[    1.478900] IR RC6 protocol handler initialized
[    1.478917] IR JVC protocol handler initialized
[    1.478935] IR Sony protocol handler initialized
[    1.478953] IR SANYO protocol handler initialized
[    1.478971] IR Sharp protocol handler initialized
[    1.478989] IR MCE Keyboard/mouse protocol handler initialized
[    1.479007] IR XMP protocol handler initialized
[    1.479021] Driver for 1-wire Dallas network protocol.
[    1.482061] poweroff-gpio gpio-poweroff: GPIO lookup for consumer (null)
[    1.482091] poweroff-gpio gpio-poweroff: using device tree for GPIO lookup
[    1.482197] of_get_named_gpiod_flags: parsed 'gpios' property of node '/gpio-poweroff[0]' - status (0)
[    1.492417] omap_wdt: OMAP Watchdog Timer Rev 0x01: initial timeout 60 sec
[    1.499161] sdhci: Secure Digital Host Controller Interface driver
[    1.499187] sdhci: Copyright(c) Pierre Ossman
[    1.502808] omap_hsmmc 48060000.mmc: GPIO lookup for consumer cd
[    1.502835] omap_hsmmc 48060000.mmc: using device tree for GPIO lookup
[    1.502904] of_get_named_gpiod_flags: parsed 'cd-gpios' property of node '/ocp/mmc@48060000[0]' - status (0)
[    1.502955] omap_hsmmc 48060000.mmc: Got CD GPIO
[    1.502989] omap_hsmmc 48060000.mmc: GPIO lookup for consumer wp
[    1.503006] omap_hsmmc 48060000.mmc: using device tree for GPIO lookup
[    1.503032] of_get_named_gpiod_flags: can't parse 'wp-gpios' property of node '/ocp/mmc@48060000[0]'
[    1.503055] of_get_named_gpiod_flags: can't parse 'wp-gpio' property of node '/ocp/mmc@48060000[0]'
[    1.503074] omap_hsmmc 48060000.mmc: using lookup tables for GPIO lookup
[    1.503103] omap_hsmmc 48060000.mmc: lookup for GPIO wp failed
[    1.509375] omap_hsmmc 47810000.mmc: GPIO lookup for consumer cd
[    1.509404] omap_hsmmc 47810000.mmc: using device tree for GPIO lookup
[    1.510215] of_get_named_gpiod_flags: can't parse 'cd-gpios' property of node '/ocp/mmc@47810000[0]'
[    1.510250] of_get_named_gpiod_flags: can't parse 'cd-gpio' property of node '/ocp/mmc@47810000[0]'
[    1.510273] omap_hsmmc 47810000.mmc: using lookup tables for GPIO lookup
[    1.510298] omap_hsmmc 47810000.mmc: lookup for GPIO cd failed
[    1.510332] omap_hsmmc 47810000.mmc: GPIO lookup for consumer wp
[    1.510349] omap_hsmmc 47810000.mmc: using device tree for GPIO lookup
[    1.510372] of_get_named_gpiod_flags: can't parse 'wp-gpios' property of node '/ocp/mmc@47810000[0]'
[    1.510395] of_get_named_gpiod_flags: can't parse 'wp-gpio' property of node '/ocp/mmc@47810000[0]'
[    1.510413] omap_hsmmc 47810000.mmc: using lookup tables for GPIO lookup
[    1.510433] omap_hsmmc 47810000.mmc: lookup for GPIO wp failed
[    1.554593] sdhci-pltfm: SDHCI platform and OF driver helper
[    1.555458] ledtrig-cpu: registered to indicate activity on CPUs
[    1.558663] omap_hsmmc 47810000.mmc: card claims to support voltages below defined range
[    1.568866] usbcore: registered new interface driver usbhid
[    1.568892] usbhid: USB HID core driver
[    1.573798] ft-TXT|init_display:: fa30e014(14) fa30e018(18)
[    1.585493] mmc0: new high speed SDIO card at address 0001
[    1.720942] Console: switching to colour frame buffer device 30x40
[    1.722156] graphics fb0: fb_ili9341_txt frame buffer, 240x320, 150 KiB video memory, 4 KiB buffer memory, fps=10
[    1.727397] oprofile: using arm/armv7
[    1.728028] Netfilter messages via NETLINK v0.30.
[    1.728913] nf_conntrack version 0.5.0 (4096 buckets, 16384 max)
[    1.729287] ctnetlink v0.93: registering with nfnetlink.
[    1.731928] ip_tables: (C) 2000-2006 Netfilter Core Team
[    1.733320] Initializing XFRM netlink socket
[    1.733884] NET: Registered protocol family 10
[    1.738512] Segment Routing with IPv6
[    1.739085] sit: IPv6, IPv4 and MPLS over IPv4 tunneling driver
[    1.743406] NET: Registered protocol family 17
[    1.743517] NET: Registered protocol family 15
[    1.745511] Key type dns_resolver registered
[    1.745793] omap_voltage_late_init: Voltage driver support not added
[    1.754042] ThumbEE CPU extension supported.
[    1.754101] Registering SWP/SWPB emulation handler
[    1.789190] omap-gpmc 50000000.gpmc: GPMC revision 6.0
[    1.789749] gpmc_mem_init: disabling cs 0 mapped at 0x0-0x1000000
[    1.789988] gpiochip_find_base: found new base at 510
[    1.791895] gpio gpiochip4: (omap-gpmc): added GPIO chardev (254:4)
[    1.792823] gpiochip_setup_dev: registered GPIOs 510 to 511 on device: gpiochip4 (omap-gpmc)
[    1.795708] omap2-nand 8000000.nand: GPIO lookup for consumer rb
[    1.795750] omap2-nand 8000000.nand: using device tree for GPIO lookup
[    1.795788] of_get_named_gpiod_flags: can't parse 'rb-gpios' property of node '/ocp/gpmc@50000000/nand@0,0[0]'
[    1.795815] of_get_named_gpiod_flags: can't parse 'rb-gpio' property of node '/ocp/gpmc@50000000/nand@0,0[0]'
[    1.795836] omap2-nand 8000000.nand: using lookup tables for GPIO lookup
[    1.795965] omap2-nand 8000000.nand: lookup for GPIO rb failed
[    1.796305] nand: device found, Manufacturer ID: 0x01, Chip ID: 0xf1
[    1.796323] nand: AMD/Spansion S34ML01G1
[    1.796345] nand: 128 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
[    1.796495] nand: using OMAP_ECC_BCH8_CODE_HW ECC scheme
[    1.796978] 11 ofpart partitions found on MTD device omap2-nand.0
[    1.796998] Creating 11 MTD partitions on "omap2-nand.0":
[    1.797435] 0x000000000000-0x000000020000 : "SPL1"
[    1.809001] 0x000000020000-0x000000040000 : "SPL2"
[    1.816758] 0x000000040000-0x000000060000 : "SPL3"
[    1.823659] 0x000000060000-0x000000080000 : "SPL4"
[    1.831354] 0x000000080000-0x0000000c0000 : "dtb"
[    1.838535] 0x0000000c0000-0x0000001c0000 : "U-boot"
[    1.846802] 0x0000001c0000-0x0000001e0000 : "U-boot-env"
[    1.853242] 0x0000001e0000-0x000000200000 : "U-boot-env-backup"
[    1.860656] 0x000000200000-0x000000700000 : "uImage"
[    1.872822] 0x000000700000-0x000000740000 : "bootlogo"
[    1.880762] 0x000000740000-0x000008000000 : "rootfs"
[    2.055534] tps65910-gpio tps65910-gpio: ti,en-gpio-sleep not specified
[    2.055695] gpiochip_find_base: found new base at 504
[    2.057095] gpio gpiochip5: (tps65910): added GPIO chardev (254:5)
[    2.057857] gpiochip_setup_dev: registered GPIOs 504 to 509 on device: gpiochip5 (tps65910)
[    2.064824] vrtc: supplied by vbat
[    2.072298] vio: supplied by vbat
[    2.076886] vdd1: supplied by vbat
[    2.081509] vdd2: supplied by vbat
[    2.089435] vdig1: supplied by vbat
[    2.093403] vdig2: supplied by vbat
[    2.097317] vpll: supplied by vbat
[    2.101513] vdac: supplied by vbat
[    2.105631] vaux1: supplied by vbat
[    2.109361] vaux2: supplied by vbat
[    2.113483] vaux33: supplied by vbat
[    2.117761] vmmc: supplied by vbat
[    2.121903] vbb: supplied by vbat
[    2.140509] tps65910-rtc tps65910-rtc: registered as rtc1
[    2.142411] omap_i2c 44e0b000.i2c: bus 0 rev0.11 at 400 kHz
[    2.148565] omap_i2c 4802a000.i2c: bus 1 rev0.11 at 400 kHz
[    2.150800] omap_hsmmc 48060000.mmc: GPIO lookup for consumer cd
[    2.150827] omap_hsmmc 48060000.mmc: using device tree for GPIO lookup
[    2.150901] of_get_named_gpiod_flags: parsed 'cd-gpios' property of node '/ocp/mmc@48060000[0]' - status (0)
[    2.151043] omap_hsmmc 48060000.mmc: Got CD GPIO
[    2.151074] omap_hsmmc 48060000.mmc: GPIO lookup for consumer wp
[    2.151092] omap_hsmmc 48060000.mmc: using device tree for GPIO lookup
[    2.151118] of_get_named_gpiod_flags: can't parse 'wp-gpios' property of node '/ocp/mmc@48060000[0]'
[    2.151144] of_get_named_gpiod_flags: can't parse 'wp-gpio' property of node '/ocp/mmc@48060000[0]'
[    2.151163] omap_hsmmc 48060000.mmc: using lookup tables for GPIO lookup
[    2.151195] omap_hsmmc 48060000.mmc: lookup for GPIO wp failed
[    2.198864] of_get_named_gpiod_flags: parsed 'gpios' property of node '/gpio_buttons@0/switch@1[0]' - status (0)
[    2.201233] input: gpio_buttons@0 as /devices/platform/gpio_buttons@0/input/input1
[    2.211043] tps65910-rtc tps65910-rtc: setting system clock to 2000-01-01 00:00:00 UTC (946684800)
[    2.226419] Freeing unused kernel memory: 2048K
[    2.248593] mmc1: host does not support reading read-only switch, assuming write-enable
[    2.250698] mmc1: new high speed SDHC card at address 0001
[    2.260077] mmcblk1: mmc1:0001 00000 29.8 GiB 
[    2.267440]  mmcblk1: p1 p2
total 0
drwxr-xr-x    2 0        0                0 Dec  8  2018 ␛[1;34m.␛[m
drwxr-xr-x   10 0        0                0 Jan  1 00:00 ␛[1;34m..␛[m
crw-------    1 0        0           5,   1 Jan  1 00:00 ␛[1;35mconsole␛[m
crw-rw-rw-    1 0        0           1,   3 Dec  8  2018 ␛[1;35mnull␛[m
crw-------    1 0        0         252,   0 Dec  8  2018 ␛[1;35mttyO0␛[m
[    4.356205] Kernel panic - not syncing: Attempted to kill init! exitcode=0x00000000
[    4.356205] 
[    4.366017] ---[ end Kernel panic - not syncing: Attempted to kill init! exitcode=0x00000000
[    4.366017] 
```

This is the complete boot log. Maybe this can help

rkunze commented:
> I´m currently thinking about integrating this:
> 
> ```
> -BR2_LINUX_KERNEL_CUSTOM_REPO_URL="git://git.ti.com/ti-linux-kernel/ti-linux-kernel.git"
> -BR2_LINUX_KERNEL_CUSTOM_REPO_VERSION="ti-linux-4.14.y"
> +BR2_LINUX_KERNEL_CUSTOM_REPO_URL="git://git.ti.com/processor-sdk/processor-sdk-linux.git"
> +BR2_LINUX_KERNEL_CUSTOM_REPO_VERSION="processor-sdk-linux-4.14.y"
> ```

Should not make much difference for now - the processor-sdk is the complete, released SDK from TI, while ti-linux-kernel is the development kernel repo from TI. The main difference for us is that ti-linux-kernel is updated more often than processor-sdk.

I'd rather switch to a mainline kernel from kernel.org instead, and follow releases there instead of always having to figure out which TI commit corresponds to which mainline point release (or just blindly following the TI stable branch, as we are doing right now). I'll try and see if a mainline kernel works for us as soon as we got the TI 4.14 kernel working, and if it does I'll switch over to mainline.

ski7777 commented:
Ok, I will try to build this:

```
diff --git a/configs/fischertechnik_TXT_defconfig b/configs/fischertechnik_TXT_defconfig
index 3d1c3001fc..58103a1b9f 100644
--- a/configs/fischertechnik_TXT_defconfig
+++ b/configs/fischertechnik_TXT_defconfig
@@ -58,9 +58,6 @@ BR2_PACKAGE_HOST_LINUX_HEADERS_CUSTOM_4_14=y
 
 # kernel
 BR2_LINUX_KERNEL=y
-BR2_LINUX_KERNEL_CUSTOM_GIT=y
-BR2_LINUX_KERNEL_CUSTOM_REPO_URL="git://git.ti.com/ti-linux-kernel/ti-linux-kernel.git"
-BR2_LINUX_KERNEL_CUSTOM_REPO_VERSION="ti-linux-4.14.y"
 BR2_LINUX_KERNEL_USE_CUSTOM_CONFIG=y
 BR2_LINUX_KERNEL_CUSTOM_CONFIG_FILE="$(BR2_EXTERNAL_FTCOMMUNITY_TXT_PATH)/board/fischertechnik/TXT/tisdk_am335x-fischertechnik_txt_defconfig"
 BR2_LINUX_KERNEL_UIMAGE=y
```

rkunze commented:
> This is the complete boot log. Maybe this can help

Yes, that matches the boot log I have. As you can see here

> ```
> [    2.248593] mmc1: host does not support reading read-only switch, assuming write-enable
> [    2.250698] mmc1: new high speed SDHC card at address 0001
> [    2.260077] mmcblk1: mmc1:0001 00000 29.8 GiB 
> [    2.267440]  mmcblk1: p1 p2
> ```

the SD card gets detected as `mmcblkp1`. And since we're expecting it to be `mmcblkp0`, mounting fails.

I'm currently trying to figure out the cause for this line:
> ` [    1.585493] mmc0: new high speed SDIO card at address 0001`
Which is my current chief suspect why we don't find the SD card at mmcblkp0 as expected.

ski7777 commented:
But mounting from the new expected path does not work:


> > This is the complete boot log. Maybe this can help
> 
> Yes, that matches the boot log I have. As you can see here
> 
> > ```
> > [    2.248593] mmc1: host does not support reading read-only switch, assuming write-enable
> > [    2.250698] mmc1: new high speed SDHC card at address 0001
> > [    2.260077] mmcblk1: mmc1:0001 00000 29.8 GiB 
> > [    2.267440]  mmcblk1: p1 p2
> > ```
> 
> the SD card gets detected as `mmcblkp1`. And since we're expecting it to be `mmcblkp0`, mounting fails.
> 
> I'm currently trying to figure out the cause for this line:
> 
> > ` [    1.585493] mmc0: new high speed SDIO card at address 0001`
> 
> Which is my current chief suspect why we don't find the SD card at mmcblkp0 as expected.

But in /dev we still don´t see any sdcards. So we have a general sd problem

ski7777 commented:
```
diff --git a/board/fischertechnik/TXT/initramfs/init b/board/fischertechnik/TXT/initramfs/init
index 3eef94d41d..51cc21483c 100755
--- a/board/fischertechnik/TXT/initramfs/init
+++ b/board/fischertechnik/TXT/initramfs/init
@@ -24,8 +24,8 @@ if ! [ -b /dev/mmcblk0p2 ] ; then
 fi
 
 # mount SD card partitions.
-mount -t vfat /dev/mmcblk0p1 /media/sdcard/boot -o dmask=022,fmask=133,flush
-mount -t ext4 /dev/mmcblk0p2 /media/sdcard/data
+mount -t vfat /dev/mmcblk1p1 /media/sdcard/boot -o dmask=022,fmask=133,flush
+mount -t ext4 /dev/mmcblk1p2 /media/sdcard/data
 mount -t squashfs /media/sdcard/boot/rootfs.img /media/sdcard/root -o loop,ro
 
 ROM_VERSION=$(cat /media/sdcard/root/etc/fw-ver.txt)

```

fails with:

```
mount: mounting /dev/mmcblk0p1 on /media/sdcard/boot failed: No such file or directory
mount: mounting /media/sdcard/boot/rootfs.img on /newroot failed: No such file or directory

```

ski7777 commented:
Ok, boots but then fails somehow. I will upload a log

ski7777 commented:
Boot works!

ski7777 commented:
https://github.com/ski7777/ftcommunity-TXT/releases/tag/v0.9.5-dev.snapshot-20181208

ski7777 commented:
```
diff --git a/board/fischertechnik/TXT/am335x-kno_txt.dts b/board/fischertechnik/TXT/am335x-kno_txt.dts
index 8c7c43582a..7ff1993644 100755
--- a/board/fischertechnik/TXT/am335x-kno_txt.dts
+++ b/board/fischertechnik/TXT/am335x-kno_txt.dts
@@ -76,6 +76,19 @@
                gpios = <&gpio1 17 GPIO_ACTIVE_HIGH>;
        };
 
+
+       kim {
+               compatible = "kim";
+               nshutdown_gpio = <97>;  /* Bank3, pin1 */
+               dev_name = "/dev/ttyO3";
+               flow_cntrl = <1>;
+               baud_rate = <3000000>;
+       };
+
+       btwilink {
+               compatible = "btwilink";
+       };
+
 };
 
 &am33xx_pinmux {
```

Now trying to add wlan and bt again fails:

```
Starting kernel ...

[    0.000990] timer_probe: no matching timers found
[    1.388657] Unable to handle kernel NULL pointer dereference at virtual address 00000000
[    1.398000] pgd = c0004000
[    1.401321] [00000000] *pgd=00000000
[    1.405312] Internal error: Oops: 5 [#1] PREEMPT ARM
[    1.410524] Modules linked in:
[    1.413746] CPU: 0 PID: 1 Comm: swapper Not tainted 4.14.79 #7
[    1.419857] Hardware name: Generic AM33XX (Flattened Device Tree)
[    1.426245] task: cd83ad00 task.stack: cd83c000
[    1.431010] PC is at kim_probe+0x78/0x214
[    1.435229] LR is at __lockdep_init_map+0x38/0x214
[    1.440250] pc : [<c0592494>]    lr : [<c016ff64>]    psr: 60000013
[    1.446817] sp : cd83de60  ip : 00000000  fp : c0c64b98
[    1.452292] r10: c0c004c8  r9 : 00000000  r8 : 00000000
[    1.457768] r7 : c1620344  r6 : c0ea7c5c  r5 : cd9c7c00  r4 : cda89500
[    1.464608] r3 : cdf4c800  r2 : c0591d60  r1 : c0ac83d8  r0 : 00000000
[    1.471450] Flags: nZCv  IRQs on  FIQs on  Mode SVC_32  ISA ARM  Segment none
[    1.478928] Control: 10c5387d  Table: 80004019  DAC: 00000051
[    1.484950] Process swapper (pid: 1, stack limit = 0xcd83c210)
[    1.491064] Stack: (0xcd83de60 to 0xcd83e000)
[    1.495643] de60: c161ff54 c0ea7c5c cd9c7c10 ffffffed c0ea7c5c fffffdfb c0ea7c5c c056a590
[    1.504220] de80: cd9c7c10 c161ff54 00000000 00000000 c0ea7c5c c056872c cd9c7c10 c0ea7c5c
[    1.512797] dea0: cd9c7c44 00000000 000000fd c0c52850 c0c004c8 c05688b4 00000000 c0ea7c5c
[    1.521374] dec0: c05687fc c0566cd8 cd886ea0 cd93b950 c0ea7c5c cdf4aa80 c0ea5fd8 c0567cd0
[    1.529951] dee0: c0ac83d8 00000000 c0c38fbc c0ea7c5c 00000000 c0c38fbc c0ee9000 c0569740
[    1.538528] df00: ffffe000 00000000 c0c38fbc c0101c90 00000000 cfdffca7 00000000 cfdffc9e
[    1.547104] df20: c0bb5210 000000fd cfdffcac c0153688 c0b40dd8 00000006 00000006 00000000
[    1.555680] df40: cfdffca7 00000000 c0c64644 00000007 c0c5286c c0ee9000 00000007 c0c52870
[    1.564256] df60: c0ee9000 c0ee9000 000000fd c0c00d6c 00000006 00000006 00000000 c0c004c8
[    1.572831] df80: 00000000 00000000 c08e39bc 00000000 00000000 00000000 00000000 00000000
[    1.581407] dfa0: 00000000 c08e39c4 00000000 c0107e94 00000000 00000000 00000000 00000000
[    1.589984] dfc0: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
[    1.598559] dfe0: 00000000 00000000 00000000 00000000 00000013 00000000 00000000 00000000
[    1.607161] [<c0592494>] (kim_probe) from [<c056a590>] (platform_drv_probe+0x50/0xac)
[    1.615378] [<c056a590>] (platform_drv_probe) from [<c056872c>] (driver_probe_device+0x208/0x2d8)
[    1.624683] [<c056872c>] (driver_probe_device) from [<c05688b4>] (__driver_attach+0xb8/0xbc)
[    1.633534] [<c05688b4>] (__driver_attach) from [<c0566cd8>] (bus_for_each_dev+0x78/0x9c)
[    1.642111] [<c0566cd8>] (bus_for_each_dev) from [<c0567cd0>] (bus_add_driver+0x198/0x1fc)
[    1.650777] [<c0567cd0>] (bus_add_driver) from [<c0569740>] (driver_register+0x78/0xf4)
[    1.659174] [<c0569740>] (driver_register) from [<c0101c90>] (do_one_initcall+0x3c/0x17c)
[    1.667763] [<c0101c90>] (do_one_initcall) from [<c0c00d6c>] (kernel_init_freeable+0x100/0x1b4)
[    1.676897] [<c0c00d6c>] (kernel_init_freeable) from [<c08e39c4>] (kernel_init+0x8/0x110)
[    1.685482] [<c08e39c4>] (kernel_init) from [<c0107e94>] (ret_from_fork+0x14/0x20)
[    1.693424] Code: 1a000053 e5943094 e59f1160 e58340ec (e5980000) 
[    1.699992] ---[ end trace 8a6278066561dffc ]---
[    1.704969] Kernel panic - not syncing: Attempted to kill init! exitcode=0x0000000b
[    1.704969] 
[    1.714563] ---[ end Kernel panic - not syncing: Attempted to kill init! exitcode=0x0000000b
[    1.714563] 

```

rkunze commented:
@ski7777 Just pushed a configuration with working WLAN here: https://github.com/rkunze/ftcommunity-TXT/tree/update-buildroot-2018.11 - can you pull it in?

Working on Bluetooth right now, and then I'll try with a mainline kernel.

ski7777 commented:
Works! I will try this with mainline already. Hopefully it will work.

ski7777 commented:
WORKS!

rkunze commented:
Great to see that it works with a mainline kernel. :+1: 

But your change pulls in the default Buildroot kernel, which is 4.18.20. This is the previous stable kernel (current stable kernel is 4.19.8), and it is already EOL.

I'll change this to use the latest mainline longterm kernel instead (currently 4.14.87), that way we don't have to chase kernel versions every 6 weeks, but we do get security and stability patches.

ski7777 commented:
Ok
The sdcard naming thing: May it be possible that the sdcard is recognized as card 1 because the sdio for the wifi is unconfigured?

rkunze commented:
No, the wifi sdio is configured (otherwise, wifi would not work - the CPU talks to the Wifi part of the wl1831 via this sdio). 

It's just that the device node for this mmc host (named "mmc3" in the DTS, just to confuse matters :grin: ), did not show up with the 4.1 config and so the second mmc host ("mmc1" in the DTS, the one connected to the sd card slot) got assigned id mmc0. I'm not sure why, but I'm not inclined to waste time chasing this down now that we've change the mount device to the new id.

ski7777 commented:
Does Bluetooth already work?

rkunze commented:
I'll know after the build is finished...

ski7777 commented:
Do you always do a full rebuild?

rkunze commented:
No, but in this case I had to pull in a new kernel.

ski7777 commented:
TBH I did all this without any clean build. A new kernel config, even a new kernel version, is a problem. Just do `make linux-dirclean all`. This is 100% clean but will it will work

rkunze commented:
I'm doing dirty builds as well (userland doesn't need to change just because I'm fiddling with the kernel). But switching from the TI kernel to mainline means I first have to pull in around 2GB of data from kernel.org, and then recompile the kernel. And both takes time, especially on the old Lenovo T500 laptop I'm using here...

rkunze commented:
https://github.com/rkunze/ftcommunity-TXT/tree/update-buildroot-2018.11 now has a kernel.org longterm kernel and working bluetooth.

@ski7777 can you pull it in here, and then test if everything else (especially ftrobopy and the FT GUI) still works with the Buildroot update and the new kernel? I'll merge once you have confirmed that we did not break anything.

ski7777 commented:
Everything works fine!

ski7777 commented:
https://github.com/ski7777/ftcommunity-TXT/releases/tag/v0.9.5-dev.snapshot-20181209

rkunze commented:
Great, thank you.
